receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -0,0 +1,147 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/textparse"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type metricFamilyPdata struct {
+	metricFamily
+	mtype  pdata.MetricDataType
```

Comment:
might be useful to comment here that we are overriding fields in metricFamily, since that has implications beyond what is seen in this file if i'm reading it correctly.  Same for metricGroupPdata below

Reply:
Done, thanks!